### DISCUSSION TITLE
FlyHigher_Tracker

### DISCUSSION THREAD

**Discussion by mth-farias** [neurogears/BonFly#2] [General]
The FlyHigherTracker.zip is missing the tensorflow.dll and .lib because it was too big.

Core nodes:

- StimuliCSV - For each experimental videos I have an aligment .csv file with the same name, each row is corresponding to each frame. This file constains the information about the different stimuli used in the experiment
- Background - This node is the first to run, it calculates the background in X number of frames and takes the last image
- QuadrantsROI - Based on the video size it divides it into quadrants
- Transform - Subtracts the background from the video and equalizes the background image so that the arena threshold is more generalizable between the different setups
- Quads - It crops the video into quadrants, isolating each individual arena
- Compression - Finds the arena Contour.Rect and crops the video to that ROI
- SLEAPTracker - SLEAPanalysis
1. Centroid - Finds the fly centroid in two ways, SoftCentroid is a high threshold value and HardCentroid is a low threshold value that finds the fly even if it jumps. BestCentroid is using the SoftCentroid unless NaN, in that case it uses the HardCentroid.
2. Zoomer - AdaptativeROI centered on the fly, this is what SLEAP was trained on spare the GPU
3. SLEAP - L,R,T,B - Is the view point to determine the fly position. The best view selects the view with higher confidence. CleanView - Can disentangle is the fly is side or top view from the fly being on an angle or vertical. Horizon is the major axis of the fly. 
4. SLEAPvisualizer - Draws the points after processing the raw data that SLEAP outputs
5. StableZoomer - Same as zoomer but it stabilizes the centroid before to clean the noise
6. PxChange - Motion in the StableZoomer (5 and 6 may be combined to calculate the motion in the whole arena, needs testing)
7. NormalizeSLEAP - Convert the X,Y values from pixel space to mm
8. OutputSLEAP - Naming the different columns

**Comment by mth-farias** [comment]
- SLEAPsavefiles

1. Save a .png image of the arena background
2. save a .csv file of the tracking output
3. Save a .avi of the video without background
- PaintEvent - Draws a visual indication of the aligment.csv to the video without background


SimpleTracker

- Simplified version of the SLEAP tracker without sleap, it only outputs centroid and motion


TestData example can be found in this drive folder:
https://drive.google.com/drive/folders/1lLnxHn0kyGBJnEw2yv6H1asiJUiWSpgN?usp=sharing

**Comment by bruno-f-cruz** [comment]
This looks great!
Unfortunately, I don't see a workflow file. Am I missing something?

**Reply by mth-farias** [reply]
Yes, we were missing something, I forgot to upload it.
Just did it :)

**Reply by bruno-f-cruz** [reply]
I am going to post it here :)
[FlyHigherTracker.zip](https://github.com/neurogears/BonFly/files/11442217/FlyHigherTracker.zip)
